acceptance/setup/git/000_register_puppetres_dll.rb

```diff
@@ -0,0 +1,12 @@
+test_name "Register puppetres.dll on any windows agents" do
+
+  agents.each do |host|
+    next unless host['platform'] =~ /windows/
+    app_name = 'HKLM\\\\SYSTEM\\\\CurrentControlSet\\\\services\\\\eventlog\\\\Application\\\\Puppet'
+    puppetres_dll_cygpath = on(host, "find /cygdrive/c -type f -name puppetres.dll").stdout.chomp
```

Comment:
It could take awhile to scan the C drive. Since `puppetres.dll` is in the puppet repo, does the install step know where it the repo is on disk, so we can use a hard-coded path like:

```ruby
"#{puppet_repo}/ext/windows/eventlog/puppetres.dll"
```

Reply:
I was originally hard-coding it, but I realized, I'm not sure of a good way to pass around `puppet_repo`. We find that in the pre-suite test by changing into the tmp directory that has the Gemfile, then running `bundle show puppet`.  I suppose we find that tmp directory by parsing the binstub bundler installs into the bindir, but beyond that, I wasn't sure the best way to find that information :(

Reply:
How about if beaker-puppet's presuite saves it in the host object?

```ruby
puppet_bundler_install_dir = on(host, "cd #{puppet_dir} && cmd.exe /c bundle show puppet").stdout.chomp
host[:puppet_bundler_install_dir] = puppet_bundler_install_dir
```

Alternatively, the only acceptance that relies on the eventlog is `acceptance/tests/windows/eventlog.rb`. We could move that test to puppet-agent, as the MSI installs the message resource dll for us.